CMakeLists.txt

```diff
@@ -26,80 +42,83 @@ target_link_libraries(mbed-baremetal
 )
 # Validate selected C library type
 # The C library type selected has to match the library that the target can support
-if(${MBED_C_LIB} STREQUAL "small")
-    if(NOT "small" IN_LIST MBED_TARGET_SUPPORTED_C_LIBS)
-        if("std" IN_LIST MBED_TARGET_SUPPORTED_C_LIBS)
-            message(WARNING
-                "We noticed that target.c_lib is set to `${MBED_C_LIB}`."
-                " As the ${MBED_TARGET} target does not support a small C library for the ${MBED_TOOLCHAIN} toolchain,"
-                " we are using the standard C library instead."
-            )
-            set(MBED_C_LIB "std" CACHE STRING "")
+if(${CMAKE_CROSSCOMPILING})
```

Comment:
Well, this is kind of okay.

But I think `MBED_TARGET` might be more direct because it's generated by mbed-tools. If this variable is not set, then it's simply not an Mbed target. What do others think?

Reply:
As discussed, there is a possibility of `mbed_config.cmake` generated by the application and it can also build the unittest so we go with `CMAKE_CROSSCOMPILING` flag checking.